barretenberg/cpp/src/barretenberg/dsl/acir_format/multi_scalar_mul.cpp

```diff
@@ -34,7 +34,11 @@ void create_multi_scalar_mul_constraint(Builder& builder,
         infinite = bool_ct(to_field_ct(input.points[i + 2], builder));
 
         if (!has_valid_witness_assignments && !input.points[i + 2].is_constant) {
-            builder.variables[input.points[i + 2].index] = fr(1);
+            if (input.points[i + 2].is_constant) {
```

Comment:
this case is never hit because of the condition on line 36.

Reply:
oh yes I forgot to remove the condition.
Still I wonder if I should handle the constant case, since it is a constant it is not concerned by 'has_witness_assignment' and thus I am changing the circuit. I will revert this last change.

Reply:
I added support for the case is_infinite constant, but x, y witness. In that case, we can set x,y to a valid value.
When one coordinate is constante, we could also set the other to a valid value, but this case can be added in a later PR.